src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -180,6 +180,59 @@ void SensorFusion::updateWorld(
     }
 }
 
+static float computeDistanceBetweenBallAndRobot(const Robot &robot, const Ball &ball)
+{
+    Point robot_position = robot.position();
+    Point ball_position  = ball.position();
+
+    double x_difference = ball_position.x() - robot_position.x();
+    double y_difference = ball_position.y() - robot_position.y();
+
+    // using the classic distance formula
+    return static_cast<float>(
+        std::sqrt(x_difference * x_difference + y_difference * y_difference));
+}
```

Comment:
nit: You can use `Vector::length` to calculate the length of a vector, alternatively, you could use our `distance` function to calculate the distance between two points directly.

Also, for cleanliness and not needing to cast values, we should stick to using doubles wherever possible.